dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeMetadataConnector.java

```diff
@@ -148,16 +155,30 @@ protected void addSqlTasksWithInfoSchemaFallback(
     }
   }
 
-  @ForOverride
-  protected void addSingleSqlTask(
+  private void addSingleSqlTask(
       @Nonnull List<? super Task<?>> out,
-      @Nonnull Class<? extends Enum<?>> header,
       @Nonnull String format,
-      @Nonnull TaskVariant task) {
+      @Nonnull TaskVariant task,
+      @Nonnull Function<ResultSet, String[]> transformer) {
     out.add(
         new JdbcSelectTask(
                 task.zipEntryName, String.format(format, task.schemaName, task.whereClause))
-            .withHeaderClass(header));
+            .withHeaderTransformer(transformer));
+  }
+
+  @CheckReturnValue
+  private String[] transformHeaderToCamelCase(ResultSet rs, CaseFormat baseFormat) {
+    try {
+      ResultSetMetaData metaData = rs.getMetaData();
+      int columnCount = metaData.getColumnCount();
+      String[] columns = new String[columnCount];
+      for (int i = 0; i < columnCount; i++) {
+        columns[i] = baseFormat.to(CaseFormat.UPPER_CAMEL, metaData.getColumnLabel(i + 1));
```

Comment:
The conversion itself is just `String` manipulation which is very unlikely to fail. The `try-catch` here is for `SQLException`. In case that occurs, we will not able to return the original column names as well.

Reply:
ref: https://github.com/apache/commons-csv/blob/9bb8da934eb3843947977b63bcaf716335179ad2/src/main/java/org/apache/commons/csv/CSVFormat.java#L487